[rewrite] bluetooth connection

Amirneutron opened the pull request:
## Description
This is the re-implementation of the Bluetooth Connection

## Related issue(s)
closes #31

Amirneutron committed:
[rewrite] bluetooth connection

Commit:
add DeviceConnection.java

Co-authored-by: armin2118 <43929673+armin2118@users.noreply.github.com>
Co-authored-by: Masiha77 <45573379+Masiha77@users.noreply.github.com>

Commit:
add tests for DeviceList.java

ferenckata reviewed (dismissed):
Very nice job!! Personally I really like that you added tests :)

Commit:
add tests for DeviceList.java

This commit adds tests for getPairedDevices() and onActivityResult().
We use @FixMethodOrder(MethodSorters.NAME_ASCENDING) to run tests in order,
since t4checkIfBTTurnsOn() moves the app in background, which fails any tests that
are run after.
